src/driver.jl

```diff
@@ -175,6 +181,10 @@ end
         end
     end
 
+    # ensure that the returned method instance is valid in the compilation world.
+    # otherwise, `jl_create_native` won't actually emit any code.
+    @assert method_instance.def.primary_world <= job.source.world <= method_instance.def.deleted_world
```

Comment:
These seems overly strict, as it is supposed to be perfectly reasonable to run a Method in any world, irrespective of its primary_world and delete_world values (which only refer to the lookup of said Method).

Reply:
> it is supposed to be perfectly reasonable to run a Method in any world

What would that look like (in the context of GPUCompiler)? I put that there because `jl_create_native` uses the same check to decide whether to emit any code: https://github.com/JuliaLang/julia/blob/7341fb9517d290be02fdc54ae453999843a0dc7e/src/aotcompile.cpp#L332-L335